Elisp updates

AndersonTorres opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AndersonTorres committed:
update-from-overlay: cosmetical change

AndersonTorres committed:
elpa-packages: updated 2022-11-08

AndersonTorres committed:
melpa-packages: updated 2022-11-08

AndersonTorres committed:
nongnu-packages: updated 2022-11-08

AndersonTorres commented:
@brprice 

https://github.com/NixOS/nixpkgs/issues/196414#issuecomment-1305051476

AndersonTorres commented:
Oh man!

By some obscure reason MELPA is failing to view Cask

https://melpa.org/#/cask

brprice commented:
> @brprice
> 
> [#196414 (comment)](https://github.com/NixOS/nixpkgs/issues/196414#issuecomment-1305051476)

Thanks for the ping. As I understand it, this PR won't directly help with #196414, since that issue is with 22.05 and this PR is targeted at master. However, I'm not very familiar with nixos' development flow, so I may just be confused here.

AndersonTorres commented:
Well, the idea would be backport this change (using git cherrypick dark magic).

AndersonTorres committed:
emacs.pkgs.cask: init at 0.8.8

Cask was ejected from MELPA (at the Cask owner's request). Now it lives as a
typical external package.

AndersonTorres committed:
manual-packages.nix: cosmetical refactor

sternenseemann commented on pkgs/top-level/all-packages.nix:
```diff
@@ -16533,8 +16533,6 @@ with pkgs;
 
   cadre = callPackage ../development/tools/cadre { };
 
-  cask = callPackage ../development/tools/cask { };
```
Why remove the top level attribute? This would at least need a changelog entry, but does it even make sense?

AndersonTorres replied:
IMHO it makes no sense to promote Cask to top-level, because it is utterly tied to Emacs. Indeed, the old expression explicitly called the internal structure of our code generation.

But I can put this in `aliases.nix`.

sternenseemann commented on pkgs/applications/editors/emacs/elisp-packages/cask/default.nix:
```diff
@@ -30,27 +54,27 @@ stdenv.mkDerivation rec {
   '';
 
+  postFixup = ''
+    touch $out/.no-upgrade
```
What is this for, can you add a comment?

AndersonTorres replied:
It came from the old expression, and it was used to avoid auto-upgrade.

Now I am feeling inclined to remove it for good.

sternenseemann reviewed (commented):
`melpa-packages.nix: cosmetical refactor` should read `manual-packages.nix`

sternenseemann commented:
@ofborg build emacs.pkgs.ada-mode

sternenseemann commented:
Okay something happened again with ada-mode's dep on wisi, I'll have to take a look.

sternenseemann commented:
Seems like ada-mode 7.1.3 (which seems to be a version not properly checked into their git repository?) needs wisi 4.1.1 which is not released on ELPA – likely also some other changes as wisi was changed somehow.

AndersonTorres committed:
elpa-packages.nix: updated 2022-11-12

AndersonTorres committed:
nongnu-packages: updated 2022-11-12

AndersonTorres committed:
melpa-packages: updated 2022-11-12

AndersonTorres commented:
@adisbladis what do you think about concentrating the manual ad-hoc packages in a subdir?
